modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/properties/CRSExtractor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * GeoTools - The Open Source Java GIS Toolkit
+ * http://geotools.org
+ *
+ * (C) 2016, Open Source Geospatial Foundation (OSGeo)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation;
+ * version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ */
+
+package org.geotools.gce.imagemosaic.properties;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.geotools.coverage.grid.io.GridCoverage2DReader;
+import org.geotools.gce.imagemosaic.properties.PropertiesCollector;
+import org.geotools.gce.imagemosaic.properties.PropertiesCollectorSPI;
+import org.geotools.referencing.CRS;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+/**
+ * Extract the CRS from the coverage in order to store in the index
+ */
+public class CRSExtractor extends PropertiesCollector {
+
+    public static final String DEFAULT_ATTRIBUTE_NAME = "crs";
+
+    private String crsCode;
+
+    public CRSExtractor() {
+        this(new CRSExtractorSPI(), Collections.singletonList(DEFAULT_ATTRIBUTE_NAME));
+    }
+
+    public CRSExtractor(PropertiesCollectorSPI spi, List<String> propertyNames) {
+        super(spi, propertyNames);
```

Comment:
The list of property names is provided because other extractors are setup with regexes and can extract multiple values (groups) in one shot, but for this one, I'd actually throw an exception if there are multiple properties, what purpose does it serve to collect the same crs value multiple times in different attributes? The same goes for DateExtractor, FSDateExtractor, they could use a "SingleValuePropertiesCollector" base class, or something like that.